platforms/core-execution/hashing/src/main/java/org/gradle/internal/hash/PrimitiveHasher.java

```diff
@@ -18,53 +18,54 @@
 
 /**
  * Hasher abstraction that can be fed different kinds of primitives that it then forwards directly to the hash function.
+ * <p>
  * Inspired by the Google Guava project – https://github.com/google/guava.
  */
 public interface PrimitiveHasher {
     /**
      * Feed a bunch of bytes into the hasher.
      */
-    void putBytes(byte[] bytes);
+    PrimitiveHasher putBytes(byte[] bytes);
```

Comment:
:x: I don't think we should update `PrimitiveHasher`. That is an unsafe thing that you shouldn't use conveniently. If you want to improve matters here, then I think we should make the class package-private.

Reply:
I wanted to do this for consistency, but I agree that it was not necessary.

Reverted this change ✔️